# Taken from https://github.com/husky/husky/blob/8cd932ac230b87ba0ce59b75908b759f4065a6c1/husky_navigation/config/costmap_exploration.yaml
track_unknown_space: true
global_frame: map
rolling_window: false
robot_base_frame: base_link
robot_radius: 0.20

plugins:
- {name: static,            type: "costmap_2d::StaticLayer"}
#BoundedExploreLayer implements several functions required to execute a frontier exploration task
- {name: explore_boundary,  type: "frontier_exploration::BoundedExploreLayer"}
#Can disable sensor layer if gmapping is fast enough to update scans
- {name: sensor,            type: "costmap_2d::ObstacleLayer"}
- {name: inflation,         type: "costmap_2d::InflationLayer"}

explore_boundary:
  resize_to_boundary: false #when receiving polygonal boundary, it resizes parent costmap accordingly
  frontier_travel_point: middle #other options closest point to robot, middle point of frontier 
  #centroid (cartesian average) of all frontier points 
  #set to false for gmapping, true if re-exploring a known area
  explore_clear_space: false #whether exploration should explore all clear space(true) or just 
  #unknown space (false)

sensor:
  observation_sources: laser
  laser: {data_type: LaserScan, clearing: true, marking: true, topic: /scan, inf_is_valid: true, raytrace_range: $(arg sensor_range), obstacle_range: $(arg sensor_range)}

inflation:
  inflation_radius: 0.05 #0.30